lxd-agent/main_agent.go

```diff
@@ -185,15 +185,13 @@ func (c *cmdAgent) Run(cmd *cobra.Command, args []string) error {
 		}
 	}()
 
-	// Only start the devlxd listener if instance-data is present.
-	if shared.PathExists("instance-data") {
-		go func() {
-			err := servers["devlxd"].Serve(devlxdListener)
-			if err != nil {
-				errChan <- err
-			}
-		}()
-	}
+	// Start the devlxd listener.
+	go func() {
+		err := servers["devlxd"].Serve(devlxdListener)
```

Comment:
Is this correct that we provide the devlxd listener in the VM always now? cc @stgraber ? 
Presumably requests to it will fail if not enabled on the LXD side, but this seems different to the presentation we have for containers?

Reply:
Yeah, it'd be nicer if we could have it turn on or off based on security.devlxd in a similar way to what we have for instances.

Given this interface won't work if not connected to the host, we could make LXD give us the value of security.devlxd during its PUT at which point we'd activate it. Similar in Update() on the LXD side, if it gets disabled, we should notify the agent through another PUT which would cause the listener to go away.